### PR TITLE
apm-data: improve indexing resilience

### PR DESCRIPTION
**Pull request by axw**
In 8.11 we were not dynamically mapping any
stack trace fields, exception attributes,
{error,transaction}.custom, cookies, or
HTTP request bodies. In 8.12 we changed to
using `flattened` in some of these and
mirrored the change in the apm-data plugin.

It has since come to light that we're seeing
indexing failures using `flattened` where
the field contents are not completely
unsanitised, e.g. in stack traces with deeply
nested stack frame variables, or source lines
that exceed 32KB.

We're reverting this use of `flattened` since
there is no requirement for these fields to
be searchable by default, and users can
override this with a custom component template.

On the other hand we are making HTTP request
and response headers `flattened`, since:
 - headers cannot be deeply nested
 - these have a natural length limit imposed by server implementations

### CODE REVIEW

**Comment by axw** [comment]
@elasticsearchmachine update branch

**Review by kyungeunni — APPROVED** [review]

File: x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_object_fields.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  - match: { hits.hits.0.fields: null }
+  - match: { hits.hits.1.fields: null }
+  - match: { hits.hits.2.fields: null }
+  - match: { hits.hits.3.fields: null }
+  - match: { hits.hits.4.fields: null }
+  - match: { hits.hits.5.fields: null }
+  - match: { hits.hits.6.fields: null }
+
+---
+"Test logs-apm.error-* flattened fields":
```
Comment (felixbarny) [inline x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_object_fields.yml:66]:
```suggestion
"Test logs-apm.error-* unmapped object fields":
```

File: x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_object_fields.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+---
+setup:
+  - do:
+      cluster.health:
+        wait_for_events: languid
+
+---
+"Test traces-apm-* object fields":
```
Comment (felixbarny) [inline x-pack/plugin/apm-data/src/yamlRestTest/resources/rest-api-spec/test/20_object_fields.yml:8]:
```suggestion
"Test traces-apm-* unmapped object fields":
```

**Review by felixbarny — APPROVED** [review]
LGTM, just an optional naming suggestion.

**Review by simitt — APPROVED** [review]
Changes LGTM

**Review by lahsivjar — APPROVED** [review]

**Pull request merged**